aviatrix/resource_aviatrix_controller_cert_domain_config.go

```diff
@@ -0,0 +1,86 @@
+package aviatrix
+
+import (
+	"context"
+	"strings"
+
+	"github.com/AviatrixSystems/terraform-provider-aviatrix/v2/goaviatrix"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAviatrixControllerCertDomainConfig() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAviatrixControllerCertDomainConfigCreate,
+		ReadContext:   resourceAviatrixControllerCertDomainConfigRead,
+		UpdateContext: resourceAviatrixControllerCertDomainConfigUpdate,
+		DeleteContext: resourceAviatrixControllerCertDomainConfigDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cert_domain": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Domain name that is used in FQDN for generating cert.",
+			},
+		},
+	}
+}
+
+func resourceAviatrixControllerCertDomainConfigCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	certDomain := d.Get("cert_domain").(string)
+
+	err := client.SetCertDomain(ctx, certDomain)
+	if err != nil {
+		return diag.Errorf("could not set cert domain: %v", err)
+	}
+
+	d.SetId(strings.Replace(client.ControllerIP, ".", "-", -1))
+	return nil
```

Comment:
Should return resourceAviatrixControllerCertDomainConfigRead

Reply:
There's only one API call in Create(), do we still need to return Read()?